src/transformers/models/bert/modeling_tf_bert.py

```diff
@@ -285,18 +184,19 @@ def call(
         assert not (input_ids is None and inputs_embeds is None)
 
         if input_ids is not None:
-            inputs_embeds = self.word_embeddings(input_ids)
+            inputs_embeds = tf.gather(params=self.weight, indices=input_ids)
+
+        input_shape = shape_list(inputs_embeds)[:-1]
 
         if token_type_ids is None:
-            input_shape = shape_list(inputs_embeds)[:-1]
             token_type_ids = tf.fill(dims=input_shape, value=0)
 
         if position_ids is None:
-            position_embeds = self.position_embeddings(inputs_embeds)
-        else:
-            position_embeds = self.position_embeddings(position_ids)
+            position_ids = tf.range(start=0, limit=input_shape[-1])[tf.newaxis, :]
```

Comment:
Can we use `tf.expand_dims(..., axis=0)` rather than using the strided slice notation? It's a numpy compatibility and not very well supported by potential ISV

Reply:
I was recently thinking the same thing, "great minds think alike." 😄